Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -329,4 +381,30 @@ ScriptExecutionContext* CookieStore::scriptExecutionContext() const
     return ActiveDOMObject::scriptExecutionContext();
 }
 
+void CookieStore::eventListenersDidChange()
+{
+    bool hadChangeEventListener = m_hasChangeEventListener;
+    m_hasChangeEventListener = hasEventListeners(eventNames().changeEvent);
+
+    if (hadChangeEventListener == m_hasChangeEventListener)
+        return;
+
+    auto* context = scriptExecutionContext();
+    if (!context)
+        return;
+
+    auto& document = *downcast<Document>(context);
+    auto* page = document.page();
+    if (!page)
+        return;
+
+    auto& cookieJar = page->cookieJar();
+    auto host = document.url().host().toString();
+
+    if (m_hasChangeEventListener)
+        cookieJar.addChangeListener(host, *this);
+    else
+        cookieJar.removeChangeListener(host, *this);
```

Comment:
We should override ActiveDOMObject::stop() and call this there too if `m_hasChangeEventListener`

Reply:
ActiveDOMObject::stop() gets called when the document is going away. No point is listening for cookie events past that point since you can't run script anymore.